src/libLookup/Lookup.cpp

```diff
@@ -2592,6 +2593,111 @@ bool Lookup::GetOfflineLookupNodes() {
   return true;
 }
 
+bool Lookup::ProcessGetDirectoryBlocksFromSeed(
+    const vector<unsigned char>& message, unsigned int offset,
+    const Peer& from) {
+  uint64_t index_num;
+  uint32_t portNo;
+
+  if (!Messenger::GetLookupGetDirectoryBlocksFromSeed(message, offset, portNo,
+                                                      index_num)) {
+    LOG_GENERAL(WARNING,
+                "Messenger::GetLookupGetDirectoryBlocksFromSeed failed");
+    return false;
+  }
+
+  vector<unsigned char> msg = {MessageType::LOOKUP,
+                               LookupInstructionType::SETDIRBLOCKSFROMSEED};
+
+  vector<boost::variant<DSBlock, VCBlock, FallbackBlockWShardingStructure>>
+      dirBlocks;
+
+  for (uint64_t i = index_num;
+       i <= m_mediator.m_blocklinkchain.GetLatestIndex(); i++) {
```

Comment:
I realized that the number of blocks in one message should be limited otherwise the message could be too long. It's easier to get message lost during transmission and may have oversize issue. However such logic has also not been implemented for GetTxBlock & GetDSBlock, which can be an optimization I think

Reply:
I think framing maybe done by underlying TCP protocol. but yeah good point.

Reply:
https://github.com/Zilliqa/Issues/issues/238